File: java-server/src/main/java/com/ditto/example/spring/quickstart/configuration/DittoMode.java
package com.ditto.example.spring.quickstart.configuration;

/**
 * Identity-mode selection based on env vars.
 *
 * <p>Non-empty {@code DITTO_OFFLINE_LICENSE_TOKEN} (after trim) selects {@link #OFFLINE};
 * otherwise the app uses {@link #ONLINE_PLAYGROUND}.
 */
public enum DittoMode {
    ONLINE_PLAYGROUND,
    OFFLINE;

    public static DittoMode select(String offlineLicenseToken) {
        if (offlineLicenseToken == null) {
            return ONLINE_PLAYGROUND;
        }
        return offlineLicenseToken.trim().isEmpty() ? ONLINE_PLAYGROUND : OFFLINE;
    }
}

File: flutter_app/lib/ditto_mode.dart
/// Identity-mode selection based on env vars.
///
/// Non-empty `DITTO_OFFLINE_LICENSE_TOKEN` (after trim) selects
/// [DittoMode.offline]; otherwise the app uses [DittoMode.onlinePlayground].
enum DittoMode { onlinePlayground, offline }

DittoMode selectDittoMode(String? offlineLicenseToken) {
  final token = offlineLicenseToken?.trim() ?? '';
  return token.isEmpty ? DittoMode.onlinePlayground : DittoMode.offline;
}
